### PR TITLE
[Catkin] Add support for cfg messages that use "parameter_generator_catkin.py" 

### PR DESCRIPTION
**Pull request by beekarthik**
I was working on a project which used cfg files that imported from "parameter_generator_catkin.py". This did not work with the provided rule and after some debugging, this is the patch developed to successfully generate language files from the cfg files.

Followed suit in terms of some of the improvements to `parameter_generator.py` made in this repo (using arg parse, short circuiting if a language gen_dir is not specified) and made corresponding changes in the 'parameter_generator_catkin.py` file

Since this is a fairly large patch, I thought it would be good to upstream.

Happy to make any changes needed to get this in! Thanks!

### CODE REVIEW

File: third_party/ros/dynamic_reconfigure/BUILD.bazel [hunk 1]
```diff
@@ -1,3 +1,4 @@
 exports_files([
     "parameter_generator.py",
+    "parameter_generator_catkin.py",
```
Comment (mvukov) [inline third_party/ros/dynamic_reconfigure/BUILD.bazel:3]: How one should use this file? For instance: `parameter_generator.py` is used in `//third_party/ros:dynamic_reconfigure` target, that's then used in `//ros:parameter_generator_app` target, finally used in `cc_ros_dynamic_reconfigure_library` rule.

Comment (beekarthik) [reply third_party/ros/dynamic_reconfigure/BUILD.bazel:3]: This new file is used in the exact same way within the `cc_ros_dynamic_reconfigure_library`. The minor variation is the import within cfg files.

In `Tutorial.cfg:1` we currently have...
```
from dynamic_reconfigure.parameter_generator import *
```

and so when `parameter_generator_app.py` [runs line 18](https://github.com/mvukov/rules_ros/blob/9b00aefe7ba84e6e22b069be461a4948f95ca32f/ros/parameter_generator_app.py#L18) it is able to find that module and import the necessary code.

In the project I am trying to migrate from catkin to bazel, the first line of every .cfg file looks like

```
from dynamic_reconfigure.parameter_generator_catkin import <objects>
```

so using the standard `cc_ros_dynamic_reconfigure_library` fails as the exec_module call in `parameter_generator_app.py` without this PR that makes that module also available. Happy to add a Tutorial2.cfg example if needed.

In an ideal world switching `parameter_generator_catkin` to `parameter_generator` in the cfg files in my project would not cause behavior changes, but it is a big legacy project and who knows what implications there are. Aiming to have Bazel reach parity with the current catkin build before diving into these kinds of changes!

Comment (mvukov) [reply third_party/ros/dynamic_reconfigure/BUILD.bazel:3]: OK, let's go with this then. Please fix the CI complaints.

Comment (mvukov) [reply third_party/ros/dynamic_reconfigure/BUILD.bazel:3]: You can install pre-commit and then the code is automatically going to be fixed when you commit code into git.

Comment (beekarthik) [reply third_party/ros/dynamic_reconfigure/BUILD.bazel:3]: sounds good - sorry about the linting idk how it slipped by.

pushed changes!!

**Review by mvukov — APPROVED** [review]

**Pull request merged**